manager/assets/modext/widgets/resource/modx.grid.trash.js

```diff
@@ -167,28 +199,33 @@ Ext.extend(MODx.grid.Trash, MODx.grid.Grid, {
         }
     },
 
-    search: function (tf, newValue) {
+    searchResource: function (tf, newValue) {
         var nv = newValue || tf;
-        this.getStore().baseParams.query = Ext.isEmpty(nv) || Ext.isObject(nv) ? '' : nv;
+        var s = this.getStore();
+        s.baseParams.query = Ext.isEmpty(nv) || Ext.isObject(nv) ? '' : nv;
+        s.load();
         this.getBottomToolbar().changePage(1);
-        this.refresh();
-        return true;
+        this.replaceState();
     },
 
     searchContext: function (tf) {
-        this.getStore().baseParams.context = !Ext.isEmpty(tf) ? tf.value : '';
+        var nv = Ext.isEmpty(tf) || Ext.isObject(tf) ? tf.value : tf;
+        var s = this.getStore();
+        s.baseParams.context = nv;
+        s.load();
         this.getBottomToolbar().changePage(1);
-        this.refresh();
-        return true;
+        this.replaceState();
     },
 
     clearFilter: function () {
-        this.getStore().baseParams.query = '';
-        this.getStore().baseParams.context = '';
-        Ext.getCmp('modx-trash-search').reset();
-        Ext.getCmp('modx-trash-context').reset();
+        var s = this.getStore();
+        s.baseParams.context = '';
+        s.baseParams.query = '';
```

Comment:
```suggestion
        s.baseParams.context = s.baseParams.query = '';
```

Reply:
I don't agree with this change request, it's more readable in
```
 s.baseParams.context = '';
s.baseParams.query = '';
```
format.